query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java

```diff
@@ -90,6 +90,7 @@ private void updateSearchFactory(final Class<?>... classes) {
          if (reducedSet.isEmpty()) {
             return;
          }
+         // TODO: may want to make this run on a different thread to make non blocking when invoking addClasses?
```

Comment:
I need input from someone who is more familiar with query if I need to move the execution of `addClasses` to another thread? It looks like it is probably a blocking operation but I couldn't quite tell.

Reply:
Going to assume this is blocking as well and change it.

Reply:
The MutableSearchFactory impl uses a lock, so I'd also say it's blocking /cc @gustavonalle @anistor

Reply:
yes, we talked about it yesterday

Reply:
This should be handled in https://github.com/infinispan/infinispan/pull/7010